public/class-local-pickup-time.php

```diff
@@ -323,17 +326,54 @@ public function create_hour_options() {
 			}
 			else {
 				// Create array of time options to return to woocommerce_form_field
-				while ( $tNow <= $tEnd ) {
+				
+				// Today
+				if ( $i == 0) {
+
+					// Check if it's not too late for pickup
+					if ( $current_time < $tEnd ) {
+
+						// Fix tNow if is pickup possible today				
+						if ( $i == 0 ) {
+							$todayStart = $tStart;
+							$delayStart = strtotime("+$delay_minutes minutes", $current_time);
+							while ( $todayStart <= $delayStart ) {
+								$todayStart = strtotime("+$interval minutes", $todayStart);
+							}
+							$tNow = $todayStart;
+						}
+
+						while ( $tNow <= $tEnd ) {
+
+							$day_name = __( 'Today', $this->plugin_slug );
 
-					$day_name = ( $i === 0 ) ? 'Today' : $current_day_name;
+							$option_key = $current_day_name . date( "_H_i", $tNow );
+							$option_value = $day_name . ' ' . date( "H:i", $tNow );
```

Comment:
I think this date format should be made available as a filter, with `date( "g:i", $tNow )` as the default (what it was set before). I know that most US based users would prefer to use 12 hour time without a leading zero.